contracts/NestedFactory.sol

```diff
@@ -271,34 +229,62 @@ contract NestedFactory is INestedFactory, ReentrancyGuard, Ownable, MixinOperato
+        uint256 batchedOrdersLength = _batchedOrders.length;
+        require(batchedOrdersLength != 0, "NF: INVALID_MULTI_ORDERS");
+        require(nestedRecords.getAssetReserve(_nftId) == address(reserve), "NF: RESERVE_MISMATCH");
 
-    /// @inheritdoc INestedFactory
-    function increaseLockTimestamp(uint256 _nftId, uint256 _timestamp) external override onlyTokenOwner(_nftId) {
-        nestedRecords.updateLockTimestamp(_nftId, _timestamp);
+        for (uint256 i = 0; i < batchedOrdersLength; i++) {
+            (uint256 fees, IERC20 tokenSold) = _submitInOrders(
+                _nftId,
+                _batchedOrders[i],
+                true,
+                _batchedOrders[i].fromReserve
+            );
+            _transferFeeWithRoyalty(fees, tokenSold, _nftId);
+        }
     }
 
-    /// @inheritdoc INestedFactory
-    function unlockTokens(IERC20 _token) external override onlyOwner {
-        uint256 amount = _token.balanceOf(address(this));
-        _token.safeTransfer(owner(), amount);
-        emit TokensUnlocked(address(_token), amount);
+    /// @dev Internal logic extraction of processOutputOrders()
+    /// @param _nftId The id of the NFT to update
+    /// @param _batchedOrders The order to execute
+    function _processOutputOrders(uint256 _nftId, BatchedOutputOrders[] calldata _batchedOrders) private {
+        uint256 batchedOrdersLength = _batchedOrders.length;
+        require(batchedOrdersLength != 0, "NF: INVALID_MULTI_ORDERS");
+        require(nestedRecords.getAssetReserve(_nftId) == address(reserve), "NF: RESERVE_MISMATCH");
+
+        for (uint256 i = 0; i < batchedOrdersLength; i++) {
+            (uint256 feesAmount, uint256 amountBought) = _submitOutOrders(
+                _nftId,
+                _batchedOrders[i],
+                _batchedOrders[i].toReserve,
+                true
+            );
+            _transferFeeWithRoyalty(feesAmount, _batchedOrders[i].outputToken, _nftId);
+            if (!_batchedOrders[i].toReserve) {
+                _safeTransferAndUnwrap(_batchedOrders[i].outputToken, amountBought - feesAmount, _msgSender());
+            }
+        }
     }
 
     /// @dev For every orders, call the operator with the calldata
     /// to submit buy orders (where the input is one asset).
     /// @param _nftId The id of the NFT impacted by the orders
     /// @param _batchedOrders The order to process
-    /// @param _reserved True if the output is store in the reserve/records, false if not.
+    /// @param _toReserve True if the output is store in the reserve/records, false if not.
     /// @param _fromReserve True if the input tokens are from the reserve
     /// @return feesAmount The total amount of fees
     /// @return tokenSold The ERC20 token sold (in case of ETH to WETH)
     function _submitInOrders(
         uint256 _nftId,
-        BatchedOrders calldata _batchedOrders,
-        bool _reserved,
+        BatchedInputOrders calldata _batchedOrders,
+        bool _toReserve,
```

Comment:
❔ Isn't `_toReserve` always true for a buy order?

Reply:
Yes! This PR will remove useless params => https://github.com/NestedFinance/nested-core-lego/pull/68